src/webui/api/appcontroller.cpp

```diff
@@ -576,6 +617,73 @@ void AppController::setPreferencesAction()
         RSS::Session::instance()->setProcessingEnabled(it.value().toBool());
     if (hasKey("rss_auto_downloading_enabled"))
         RSS::AutoDownloader::instance()->setProcessingEnabled(it.value().toBool());
+
+    // libtorrent preferences
+    // Async IO threads
+    if (hasKey("async_io_threads"))
+        session->setAsyncIOThreads(it.value().toInt());
+    // Checking Memory Usage
+    if (hasKey("checking_memory_use"))
+        session->setCheckingMemUsage(it.value().toInt());
+    // Disk write cache
+    if (hasKey("disk_cache"))
+        session->setDiskCacheSize(it.value().toInt());
+    if (hasKey("disk_cache_ttl"))
+        session->setDiskCacheTTL(it.value().toInt());
+    // Enable OS cache
+    if (hasKey("enable_os_cache"))
+        session->setUseOSCache(it.value().toBool());
+    // Guided read cache
+    if (hasKey("enable_guided_read_cache"))
+        session->setGuidedReadCacheEnabled(it.value().toBool());
+    // Coalesce reads & writes
+    if (hasKey("enable_coalesce_read_write"))
+        session->setCoalesceReadWriteEnabled(it.value().toBool());
+    // Suggest mode
+    if (hasKey("enable_upload_suggestions"))
+        session->setSuggestMode(it.value().toBool());
+    // Send buffer watermark
+    if (hasKey("send_buffer_watermark"))
+        session->setSendBufferWatermark(it.value().toInt());
+    if (hasKey("send_buffer_low_watermark"))
+        session->setSendBufferLowWatermark(it.value().toInt());
+    if (hasKey("send_buffer_watermark_factor"))
+        session->setSendBufferWatermarkFactor(it.value().toInt());
+    // Outgoing ports
+    if (hasKey("outgoing_ports_min"))
+        session->setOutgoingPortsMin(it.value().toInt());
+    if (hasKey("outgoing_ports_max"))
+        session->setOutgoingPortsMax(it.value().toInt());
+    // uTP-TCP mixed mode
+    if (hasKey("utp_tcp_mixed_mode"))
+        session->setUtpMixedMode(static_cast<BitTorrent::MixedModeAlgorithm>(it.value().toInt()));
```

Comment:
Just want to make sure that you are fully aware that static_cast won't always return a valid value.
And I think the current way is acceptable to me as I don't want any maintenance.